app/services/checkout/gateway_checkout_session.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Checkout
+  class GatewayCheckoutSession
+    def initialize(payment, success_url, cancel_url)
+      @order = payment.order
+      @payment = payment
+      @success_url = success_url
+      @cancel_url = cancel_url
+    end
+
+    attr_reader :payment
+
+    def session
+      if @order.gateway_checkout_session_id.present?
+        checkout_session = Stripe::Checkout::Session.retrieve(@order.gateway_checkout_session_id)
+      else
+        checkout_session = Stripe::Checkout::Session.create({
+          customer_email: @order.user&.email,
+          line_items: line_items,
```

Comment:
As a `checkout_session` is linked to an `order`, which is linked to `line_items`, why should we link a `checkout_session` to `line_items` too?